src/design-patterns/MazeGame/builder/standard_maze_builder.cpp

```diff
@@ -0,0 +1,61 @@
+#include "standard_maze_builder.hpp"
+
+#include <iostream>
+
+#include "door.hpp"
+#include "room.hpp"
+#include "wall.hpp"
+
+StandardMazeBuilder::StandardMazeBuilder() : currentMaze(nullptr) {
+  std::cout << "Init StandardMazeBuilder()" << std::endl;
+}
+
+std::unique_ptr<Maze> StandardMazeBuilder::GetMaze() {
+  std::cout << "StandardMazeBuilder::GetMaze()" << std::endl;
+
+  return std::move(currentMaze);
+}
+
+Direction StandardMazeBuilder::CommonWall(const Room* room1,
+                                          const Room* room2) {
+  std::cout << "StandardMazeBuilder::CommonWall(" << room1->GetRoomNumber()
+            << ", " << room2->GetRoomNumber() << ")" << std::endl;
+
+  if (room1->GetRoomNumber() < room2->GetRoomNumber()) {
+    return Direction::East;
+  } else {
+    return Direction::West;
+  }
```

Comment:
```suggestion
Direction StandardMazeBuilder::CommonWall(const Room* const room1,
                                          const Room* const room2) {
  assert (room1 != nullptr && room2 != nullptr);
  std::cout << "StandardMazeBuilder::CommonWall(" << room1->GetRoomNumber()
            << ", " << room2->GetRoomNumber() << ")" << std::endl;

  if (room1->GetRoomNumber() < room2->GetRoomNumber()) {
    return Direction::East;
  } else {
    return Direction::West;
  }
```

값 접근만 하는거면 주소값 및 주소가 가리키는 값 모두 상수화시키는게 좋지
그리고 받아온 값이 null pointer면 segfault 뜨니까 검사하는게 좋겠음

Reply:
```suggestion
Direction StandardMazeBuilder::CommonWall(const Room &room1,
                                          const Room &room2) {
  std::cout << "StandardMazeBuilder::CommonWall(" << room1->GetRoomNumber()
            << ", " << room2.GetRoomNumber() << ")" << std::endl;

  if (room1.GetRoomNumber() < room2.GetRoomNumber()) {
    return Direction::East;
  } else {
    return Direction::West;
  }
```

다 귀찮으면 레퍼로 받아오기

Reply:
어렵다 시플플

Reply:
### const Room* const room1:

이 선언은 room1이 const Room 타입을 가리키는 포인터이며, 포인터 자체도 상수라는 것을 의미합니다.
즉, room1을 통해 가리키는 Room 객체는 수정할 수 없으며(const Room), room1 포인터 자체도 다른 객체나 주소로 변경할 수 없습니다(const room1).

### const Room* room1:

여기서 room1은 const Room 타입을 가리키는 포인터입니다.
이는 room1을 통해 가리키는 Room 객체를 수정할 수 없음을 의미하지만, room1 포인터 자체는 다른 객체나 주소로 변경할 수 있습니다.

### const Room &room1:

이 선언은 Room 타입의 상수 참조를 의미합니다.
room1은 Room 객체에 대한 참조이며, 해당 객체는 수정할 수 없습니다(const Room).
참조는 한 번 초기화되면 다른 객체를 참조할 수 없으므로, room1은 초기화된 객체를 계속 참조하게 됩니다.